### PR TITLE
ci: add release asset github action

### PR DESCRIPTION
**Pull request by zemyblue**
## Description

closes: #12 
when add tag, github action build cross os binary and add as assets in release note.

### CODE REVIEW

**Review by jaeseung-bae — APPROVED** [review]

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      - name: Create a context for buildx
+        run: docker context create buildx
+
+      - name: Set up docker buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          endpoint: buildx
+          config-inline: |
+            [registry."docker.io"]
```
Comment (tkxkd0159) [inline .github/workflows/docker.yml:32]: This config doesn't affect anything. Isn't it better to remove?

Comment (zemyblue) [reply .github/workflows/docker.yml:32]: Thanks, I'll check it.

**Review by jaeseung-bae — APPROVED** [review]

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        uses: docker/login-action@v2
+        with:
+          registry: ${{env.registry}}
+          username: ${{secrets.DOCKERHUB_USERNAME}}
+          password: ${{secrets.DOCKERHUB_TOKEN}}
+
+      - name: Find go version
+        id: find_go_version
+        run: |
+          GO_VERSION=$(cat go.mod | grep -E 'go [0-9].[0-9]+' | cut -d ' ' -f 2)
```
Comment (0Tech) [inline .github/workflows/docker.yml:40]:
```suggestion
          GO_VERSION=$(grep -E '^go [0-9]\.[0-9]+' go.mod | cut -d ' ' -f 2)
```

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          echo "GO_VERSION=$GO_VERSION" >> $GITHUB_ENV
+
+      - name: Find Ostracon version
+        id: find_ostracon_version
+        run: |
+          OST_VERSION=$(go list -m github.com/Finschia/ostracon | sed 's:.* ::')
```
Comment (0Tech) [inline .github/workflows/docker.yml:46]: Is there any case which `OST_VERSION` is not an empty string?

Comment (zemyblue) [reply .github/workflows/docker.yml:46]: There will always be version information of ostracon in go.mod. But `OST_VERSION` is not automatically filled in.

Comment (0Tech) [reply .github/workflows/docker.yml:46]: Never mind. The diff's newline made me confused - I missed the space bar.

**Review by 0Tech — APPROVED** [review]
LGTM

**Pull request merged**